src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs

```diff
@@ -106,11 +119,37 @@ public static IAdalLogCallback Callback
             internal get { return _localCallback; }
         }
 
+        internal static void ExecuteCallback(LogLevel level, string message)
+        {
+            lock (LockObj)
+            {
+                _localCallback?.Log(level, message);
+            }
+        }
+
+        private static volatile LogCallback _logCallback;
+
+        /// <summary>
+        ///     Callback instance that can be provided by the developer to consume and publish logs in a custom manner.
```

Comment:
add documentation that if delegate is configured then obsolete callback will not be executed.